docs/content/docs/dev/table/sourcesSinks.md

```diff
@@ -223,6 +224,20 @@ for more information.
 The runtime implementation of a `LookupTableSource` is a `TableFunction` or `AsyncTableFunction`. The function
 will be called with values for the given lookup keys during runtime.
 
+#### Vector Search Table Source
+
+A `VectorSearchTableSource` searches an external storage system using an input vector and returns the most similar top-K rows during runtime. Users 
+can determine which algorithm to use to calculate the similarity between the input data and data stored in the external system. In general, most 
+vector databases support using Euclidean distance or Cosine distance to calculate similarity.
+
+Compared to `ScanTableSource`, a `VectorSearchTableSource` currently only supports emitting insert-only changes.
```

Comment:
I think it would be useful to explain more what this means including why are we emitting inserts for a read only search. How does this compare to the lookup processing in this respect. 

I wonder if we could put together a table with as many relevant capabilities and how they effect lookup scan and Vector Search Table. This would make it really easy to see the similarities and differences. I would suggest you  include example SQL that the planner would use to would target each of these behaviours.